<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<title>MutScan presentation</title>

		<link rel="stylesheet" href="css/reveal.css">
		<link rel="stylesheet" href="css/theme/deepblue.css">

		<!-- Theme used for syntax highlighting of code -->
		<link rel="stylesheet" href="lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>
	</head>
	<body>
		<div class="reveal">
			<div class="slides">
				<section>
					<h1><a target="_blank" href="https://github.com/OpenGene/MutScan/">MutScan</a></h1>
					<h4>fast detection and visualization of target mutations by scanning FASTQ data</h4>
					<p>
						<small>Shifu Chen, Tanxiao Huang, Tiexiang Wen, Hong Li, Mingyan Xu and Jia Gu</small>
					</p>
					<p>
						<small><a target="_blank" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2024-6">
							BMC Bioinformatics (January 2018) 19:16 DOI 10.1186/s12859-018-2024-6
						</a></small>
					</p>
					<p><small>Speaker: Miguel Gonz√°lez Sosa</small></p>
					<p><small><a href="mailto:sosa@uma.es">sosa@uma.es</a></small></p>
				</section>
				<section>
					<h2>What is it for?</h2>
					<ul>
						<li>check whether the given FASTQ files contain some specific mutations.</li>
						<li>visualize and validate them to avoid false positive calling if you have no
							enough confidence with the mutations called by your pipeline.</li>
						<li>check that in a fast way if you worry that your pipeline uses too strict
							filtering and may cause some false negative.</li>
						<li>visualize the called mutation  with its clear pipeup information
							in a more graphic way.</li>
					</ul>
				</section>
				<section>
					<h2>Why use it?</h2>
					<ul>
						<li>open source.</li>
						<li>simple to use.</li>
						<li>multi-threaded.</li>
						<li>can be 50X+ faster than normal pipeline (<a href="https://github.com/lh3/bwa">BWA</a> + <a href="https://github.com/samtools/samtools">Samtools</a> + <a href="https://github.com/broadinstitute/gatk">GATK</a>/<a href="http://varscan.sourceforge.net/">VarScan</a>/<a href="https://github.com/broadinstitute/mutect">Mutect</a>).</li>
						<li>60 built-in cancer-related target mutations to search by default.</li>
					</ul>
				</section>
				<section>
					<h2>Technologies.</h2>
					<ul>
						<li>Coded in C++.
							<img class="fragment current-visible center-image"
							src="lib/resources/proglang.png" alt="Programming languages used.">
						</li>
						<li class="fragment">Depends on <a href="https://zlib.net/"><code>zlib</code></a> library.</li>
						<li class="fragment">Doesn't support Windows natively -> Cygwin-2.6.0/g++ 5.4.</li>
					</ul>
				</section>
				<section>
					<h2>Usage.</h2>
					<pre><code class="bash" data-trim contenteditable>
						mutscan -1 R1.fq.gz -2 R2.fq.gz # pair-end
						mutscan -1 R1.fq.gz # single-end
					</code></pre>
					<p>then open <code>mutscan.html</code> with some browser and <strong>thats all!</strong></p>
					<p>We have more options, though</p>
					<pre><code class="bash" data-trim contenteditable>
						mutscan -1 R1.fq -2 R2.fq -m target.csv # or
						mutscan -1 R1.fq -2 R2.fq -m target.vcf -r hg19.fa
					</code></pre>
				</section>
				<section>
					<h4>And more advanced options</h4>
					<pre><code data-trim class="bash" contenteditable="">
usage: mutscan -1 &lt;read1_file&gt; -2 &lt;read2_file&gt; [options]...
options:
  -1, --read1                read1 file name, required
  -2, --read2                read2 file name
  -m, --mutation             mutation file name, can be a CSV format or a VCF format
  -r, --ref                  reference fasta file name (only needed when mutation file is a VCF)
  -h, --html                 filename of html report, default is mutscan.html in work directory
  -t, --thread               worker thread number, default is 4
  -S, --support              min read support required to report a mutation, default is 2.
  -k, --mark                 when mutation file is a vcf file, --mark means only process the records with FILTER column is M
  -l, --legacy               use legacy mode, usually much slower but may be able to find a little more reads in certain case
  -s, --standalone           output standalone HTML report with single file. Don't use this option when scanning too many target mutations (i.e. >1000 mutations)
  -n, --no-original-reads    dont output original reads in HTML and text output. Will make HTML report files a bit smaller
  -?, --help                 print this message
					</code></pre>
				</section>
				<section>
					<h2>Why did I choose this software?</h2>
				</section>
				<section>
					<h2>Code quality.</h2>
					<ul>
						<li><span class="pro">Modular and well-formated code</span>.</li>
						<li><span class="con">Improvable directory hierarchy</span>.</li>
						<li>Documentation: <span class="pro">user manual</span>,
							but maybe <span class="con">there could be more comments in the code</span>.</li>
						<li><i><span class="con">Handmade</span></i> <span class="pro">unit tests</span>: not standarized with any framework.</li>
						<li><span class="con">Tests and functional code are in the same class</span>.</li>
						<li><span class="con">There is no documentation related with the unit tests</span>.</li>
					</ul>
				</section>
				<section>
					<h2>Conclusions.</h2>
				</section>
			</div>
		</div>

		<script src="lib/js/head.min.js"></script>
		<script src="js/reveal.js"></script>

		<script>
			// More info about config & dependencies:
			// - https://github.com/hakimel/reveal.js#configuration
			// - https://github.com/hakimel/reveal.js#dependencies
			Reveal.configure({ pdfMaxPagesPerSlide: 1 });
			Reveal.initialize({
				dependencies: [
					{ src: 'plugin/markdown/marked.js' },
					{ src: 'plugin/markdown/markdown.js' },
					{ src: 'plugin/notes/notes.js', async: true },
					{ src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } }
				]
			});
		</script>
	</body>
</html>
